.travis.yml

```diff
@@ -1,22 +1,83 @@
 language: bash
-sudo: required
-script: bash shellcheck && sudo bash buildall
-dist: xenial
+dist: focal
+virt: vm
+group: edge
+os: linux
 services:
   - docker
-before_install:
-  - docker version
-  # Fix for Ubuntu Xenial apt-daily.service triggering
-  # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
-  - |
-    while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
-      sleep 1
-    done
-  - sudo apt-get -qq update
-  - sudo apt-get install -y debian-archive-keyring debootstrap shellcheck
-deploy:
-  provider: script
-  script: bash pushall
-  skip_cleanup: true
-  on:
-    branch: master
+
+env:
+  - BASENAME=bitnami/minideb
+
+.build_job: &build_job
+  stage: build
+  before_install:
+    - docker version
+    # Fix for Ubuntu Xenial apt-daily.service triggering
+    # https://unix.stackexchange.com/questions/315502/how-to-disable-apt-daily-service-on-ubuntu-cloud-vm-image
+    - |
+      while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do
+        sleep 1
+      done
+    - sudo rm -f /usr/local/bin/jq
+  install:
+    - sudo make .installed-requirements
+  script:
+    - sudo bash buildone $DIST $PLATFORM
+    - 'if [[ "$TRAVIS_BRANCH" == "master" && "$DIST" == "buster" ]] ; then sudo bash buildone_snapshot $DIST "$(./snapshot_id)" $PLATFORM ; fi'
+  after_success:
+    - 'if [[ "$TRAVIS_BRANCH" == "master" && "$DIST" == "buster" ]] ; then sudo docker tag "$BASENAME:$DIST-$PLATFORM" "$BASENAME:latest-$PLATFORM" ; fi'
+    - 'if [[ "$TRAVIS_BRANCH" == "master" ]] ; then sudo bash pushone $DIST $PLATFORM ; fi'
+    - 'if [[ "$TRAVIS_BRANCH" == "master" && "$DIST" == "buster" ]] ; then sudo bash pushone "buster-$(./snapshot_id)" $PLATFORM ; fi'
```

Comment:
Checking the pushall script, it seems the tag when adding a snapshot id was formed as:

`TAG="${DIST}-snapshot-${snapshot_id}"`

Unless calling the script instead of cat-ing the file is adding the prefix, I believe the line should be:

`then sudo bash pushone "buster-$(./snapshot_id)" $PLATFORM`

Also, as the "buster" dist is probably going to change in the future, we should use env vars so it can be easily modified.

In pushall we are using LATEST to point to buster to control which dist gets the "latest" tag, and `DISTS_WITH_SNAPSHOT` to control which dists should get snapshots. I'm not sure if our intention is to always make it a one-element list that has the same value as `$LATEST` though.

Reply:
I've added LATEST and DISTS_WITH_SNAPSHOT env vars to travis config and fixed the tag name.